test/powershell/Language/Scripting/ActionPreference.Test.ps1

```diff
@@ -0,0 +1,129 @@
+            }
+            catch {}
+             
+            It '$err.Count' { $err.Count | Should Be 1 }
+            It '$err[0] should not be $null' { $err[0] | Should Not Be $null }
+            It '$err[0].GetType().Name' { $err[0].GetType().Name | Should Be ActionPreferenceStopException }
+            It '$err[0].ErrorRecord' { $err[0].ErrorRecord | Should not BeNullOrEmpty }
+            It '$err[0].ErrorRecord.Exception.GetType().Name' { $err[0].ErrorRecord.Exception.GetType().Name | Should Be ItemNotFoundException }
+        }
+        
+        It 'ActionPreference Ignore Works' {            
+            $errorCount = $error.Count
+            Get-Process -Name asdfasdfsadfsadf -ErrorAction Ignore
+            $errorCount2 = $error.Count
+
+            $error.Count | Should Be $errorCount
+        }
+        
+        Context 'action preference of Ignore cannot be set as a preference variable' {            
+            $failed = $true
+            try {
+                $GLOBAL:errorActionPreference = "Ignore"
+                Get-Process -Name asdfasdfasdf
+                $failed = $false
+             } catch {
+                    It '$_.CategoryInfo.Reason' { $_.CategoryInfo.Reason | Should Be NotSupportedException }
+             }
+
+            It 'Expect exception' { $failed | Should be $true }
+        }
+
+        Context 'action preference of Suspend cannot be set as a preference variable' {
+            $failed = $true
+            try { 
+                $GLOBAL:errorActionPreference = "Suspend"
+                Get-Process -Name asdfasdfasdf
+                $failed = $false
+                } catch { 
+                    It '$_.CategoryInfo.Reason' { $_.CategoryInfo.Reason | Should Be ArgumentTransformationMetadataException }
+                }
+                It 'Expect exception' { $failed | Should be $true }
+        }
+
+        It 'enum disambiguation works' {
+            $errorCount = $error.Count
+            Get-Process -Name asdfasdfsadfsadf -ErrorAction Ig
+
+            $error.Count | Should Be $errorCount
+        }
+
+        It 'ErrorAction = Suspend works on Workflow' -Skip:$IsCoreCLR {
+           . .\TestsOnWinFullOnly.ps1
+            Run-TestOnWinFull "ActionPreference:ErrorAction=SuspendOnWorkflow"
+        }
+
+        Context 'ErrorAction = Suspend does not work on functions' {
+
+            function MyHelperFunction {
+                [CmdletBinding()]
+                param()        
+                "Hello"
+            }
+    
+            $failed = $true
+            try
+            {
+                MyHelperFunction -ErrorAction Suspend
+                $failed = $false
+            } catch {
+                It '$_.FullyQualifiedErrorId' { $_.FullyQualifiedErrorId | Should Be "ParameterBindingFailed,MyHelperFunction" }
+            }
+            It 'Expect exception' { $failed | Should be $true }
+        }
```

Comment:
but this function only be accessible within this Context. it is not used for other context.